devimint/src/util.rs

```diff
@@ -251,30 +252,23 @@ pub async fn poll_max_retries<Fut>(name: &str, retries: usize, f: impl Fn() -> F
 where
     Fut: Future<Output = Result<bool>>,
 {
-    let mut i = 0;
-    loop {
-        let result = f().await;
-        if i == retries - 1 {
-            match result {
-                Ok(true) => return Ok(()),
-                Ok(false) => {
-                    bail!("{name} failed to reach good state after {retries} retries");
-                }
-                Err(e) => {
-                    bail!("{name} failed after {retries} retries with: {e:?}");
-                }
+    for i in 0.. {
+        match f().await {
+            Ok(true) => return Ok(()),
+            other if i <= retries => {
+                debug!("polling {name} failed with: {other:?}, will retry... ({i}/{retries})");
+                task::sleep(Duration::from_secs(1)).await;
+            }
+            Ok(false) => {
+                bail!("{name} failed to reach good state after {retries} retries");
             }
-        } else {
-            match result {
-                Ok(true) => return Ok(()),
-                other => {
-                    i += 1;
-                    debug!("polling {name} failed with: {other:?}, will retry... ({i}/{retries})");
-                    task::sleep(Duration::from_secs(1)).await;
-                }
+            Err(e) => {
+                bail!("{name} failed after {retries} retries with: {e:?}");
             }
         }
     }
+
+    unreachable!();
```

Comment:
I'm pretty sure I saw that change already, did you split another PR I had already reviewed?

Reply:
I did similar change twice in two places, and also I think this commit was somewhere else already.